### PR TITLE
add support for regenerating ids

### PR DESCRIPTION
**Pull request by eth0net**
## Description

This adds support for regenerating the id for a category, entity, or property. It also adds the ability to regenerate ids for all pandora items at once, to aid with migration from sequential ids to nanoids.

## Addressed issues

- none

## Screenshots

image
image

### CODE REVIEW

**Comment by eth0net** [comment]
One flaw right now: the list won't automatically update tooltips with the new IDs. They do update after a save and reload of items using the Pandora toolbar.

**Comment by eth0net** [comment]
> One flaw right now: the list won't automatically update tooltips with the new IDs. They do update after a save and reload of items using the Pandora toolbar.

Fixed using `_populate_data()`, though this does cause the selected entry to be de-selected, which seems unnecessary.

**Comment by eth0net** [comment]
There were some whitespace changes done by Godot, so worth hiding whitespace when reviewing :)

**Comment by bitbrain** [comment]
A few things we may want to add:

- unit tests
- some basic docs of what the purpose of this functionality is (e.g. use cases, examples)

File: addons/pandora/plugin.gd [hunk 1]
```diff
@@ -15,6 +15,8 @@ func _init() -> void:
 func _enter_tree() -> void:
 	add_autoload_singleton("Pandora", "res://addons/pandora/api.gd")
 	
+	PandoraSettings.initialize()
```
Comment (bitbrain) [inline addons/pandora/plugin.gd:18]: Could we make these settings part of the project settings instead? https://docs.godotengine.org/en/stable/classes/class_projectsettings.html

**Review by bitbrain — APPROVED** [review]

**Comment by bitbrain** [comment]
Thank you! Great PR.

**Pull request merged**